src/DashboardService.Business/TaskType/EditTaskTypeCommand.cs

```diff
@@ -1,25 +1,60 @@
+using FluentValidation.Results;
 using LT.DigitalOffice.DashboardService.Business.TaskType.Interfaces;
 using LT.DigitalOffice.DashboardService.Data.Interfaces;
+using LT.DigitalOffice.DashboardService.Mappers.Edit.Interfaces;
 using LT.DigitalOffice.DashboardService.Models.Dto.Requests.TaskType;
+using LT.DigitalOffice.DashboardService.Validation.TaskType.Interfaces;
+using LT.DigitalOffice.Kernel.BrokerSupport.AccessValidatorEngine.Interfaces;
+using LT.DigitalOffice.Kernel.Helpers.Interfaces;
 using LT.DigitalOffice.Kernel.Responses;
 using Microsoft.AspNetCore.JsonPatch;
-using Microsoft.AspNetCore.Mvc;
 using System;
+using System.Linq;
+using System.Net;
 using System.Threading.Tasks;
 
 namespace LT.DigitalOffice.DashboardService.Business.TaskType;
 
 public class EditTaskTypeCommand : IEditTaskTypeCommand
 {
-  private readonly ITaskTypeRepository _repository;
-  
-  public EditTaskTypeCommand(ITaskTypeRepository repository) 
+  private readonly ITaskTypeRepository _taskTypeRepository;
+  private readonly IAccessValidator _accessValidator;
+  private readonly IResponseCreator _responseCreator;
+  private readonly IEditTaskTypeRequestValidator _validator;
+  private readonly IEditDbTaskTypeMapper _mapper;
+
+  public EditTaskTypeCommand(
+    ITaskTypeRepository repository,
+    IAccessValidator accessValidator,
+    IResponseCreator responseCreator,
+    IEditTaskTypeRequestValidator validator,
+    IEditDbTaskTypeMapper mapper)
   {
-    _repository = repository;
+    _taskTypeRepository = repository;
+    _accessValidator = accessValidator;
+    _responseCreator = responseCreator;
+    _validator = validator;
+    _mapper = mapper;
   }
   
-  public Task<OperationResultResponse<bool>> ExecuteAsync(Guid id, JsonPatchDocument<PatchTaskTypeRequest> request)
+  public async Task<OperationResultResponse<bool>> ExecuteAsync(Guid id, JsonPatchDocument<EditTaskTypeRequest> request)
   {
-    throw new NotImplementedException();
+    if (!await _accessValidator.IsAdminAsync())
+    {
+      return _responseCreator.CreateFailureResponse<bool>(HttpStatusCode.Forbidden);
+    }
+
+    ValidationResult validationResult = await _validator.ValidateAsync((id, request));
+
+    if (!validationResult.IsValid)
+    {
+      return _responseCreator.CreateFailureResponse<bool>(HttpStatusCode.BadRequest,
+        validationResult.Errors.Select(vf => vf.ErrorMessage).ToList());
```

Comment:
validationResult.Errors is List or ICollection?

Reply:
List 
![image](https://github.com/LT-Students/DashboardService/assets/42372955/7965df51-3f0e-4743-aea2-d6b67a1373e9)

Reply:
So may be change Select(vf => vf.ErrorMessage).ToList() to ConvertAll()?

Reply:
++